emscripten/webarkitJsfeat.cpp

```diff
@@ -168,25 +172,37 @@ void train_orb_pattern_internal(const char* filename) {
   imgproc.resample(img_u8.get(), lev0_img.get(), new_width, new_height);
 
   // prepare preview
-  // pattern_preview = new jsfeat.matrix_t(new_width >> 1, new_height >> 1, jsfeat.U8_t | jsfeat.C1_t);
   std::unique_ptr<Matrix_t> pattern_preview = std::make_unique<Matrix_t>(jpegImage->xsize >> 1, jpegImage->ysize >> 1, ComboTypes::U8C1_t);
   imgproc.pyrdown_internal(lev0_img.get(), pattern_preview.get());
-  
+
   Array<KeyPoints> lev_corners;
+  Array<std::unique_ptr<Matrix_t>> pattern_descriptors;
+
   for (lev = 0; lev < num_train_levels; ++lev) {
-    //pattern_corners[lev] = [];
-    //lev_corners = pattern_corners[lev];
+    // what we should do with this code?
+    // pattern_corners[lev] = [];
+    // lev_corners = pattern_corners[lev];
 
     // preallocate corners array
     i = (new_width * new_height) >> lev;
     while (--i >= 0) {
-      //lev_corners[i] = new jsfeatCpp.keypoint_t(0, 0, 0, 0, -1);
       lev_corners[lev].set_size(i);
     }
-
-   // pattern_descriptors[lev] = new jsfeatCpp.matrix_t(32, max_per_level, jsfeat.U8_t | jsfeat.C1_t);
+    pattern_descriptors.push_back(std::unique_ptr<Matrix_t>(new Matrix_t(32, max_per_level, ComboTypes::U8C1_t)));
   }
 
+  imgproc.gaussian_blur_internal(lev0_img.get(), lev_img.get(), 5, 0.2);  // this is more robust
+  corners_num = detectors.detect_keypoints(lev_img.get(), lev_corners[0], max_per_level);
+
+  // orb.describe(lev_img.get(), lev_corners[0], corners_num, lev_descr.get());
+  // This probablly will work in a near future
+  // orb.describe(lev_img.get(), lev_corners[0], corners_num, &pattern_descriptors[0]);
```

Comment:
orb.describe can not be yet used here because it accept in the first parameter a `uintptr_t` and in the second parameter an emscripten::val can not be managed here. I should creagte a new method in the **Orb** class:
`orb.describe_internal(Matrix_t* mat, Keypoints* kp, int num corners, Matix_t* descr)`